src/server/sendLayoutHTTPResponse.js

```diff
@@ -64,12 +65,17 @@ export async function sendLayoutHTTPResponse(renderOptions) {
     applicationPropPromises = {},
     headerPromises = {};
 
+  const assetsStream = merge2({
```

Comment:
To ensure that we wait for all assets to render before proceeding, I think we should add the `end: false` option, as explained at https://github.com/teambition/merge2#options. I think that that way we can keep adding to the assetsStream even after it's added to bodyStream, but still preserve correct order.

Reply:
Needed to update the CustomParser.js and add the assets tag there . This was causing it to be pushed to the body

Reply:
👍 good find. We may also need to add the `end: false` here to ensure correct ordering of the HTML. However, that is something we can verify in tests